apm-agent-plugins/apm-aws-sdk/apm-aws-sdk-2-plugin/src/main/java/co/elastic/apm/agent/awssdk/v2/helper/S3Helper.java

```diff
@@ -35,4 +38,22 @@ public static S3Helper getInstance() {
     public S3Helper(Tracer tracer) {
         super(tracer, SdkV2DataSource.getInstance());
     }
+
+    @Nullable
+    @Override
+    protected String getObjectKey(SdkRequest request, @Nullable String bucketName) {
+        String value = awsSdkDataSource.getFieldValue(IAwsSdkDataSource.OBJECT_KEY_FIELD, request);
+
+        if (value == null) {
+            value = awsSdkDataSource.getFieldValue(IAwsSdkDataSource.OBJECT_DESTINATION_KEY_FIELD, request);
+        }
+
+        return value;
+    }
+
+    @Nullable
+    @Override
+    protected String getCopySource(SdkRequest request) {
+        return awsSdkDataSource.getFieldValue(IAwsSdkDataSource.COPY_SOURCE_FIELD, request);
+    }
```

Comment:
[for reviewer] with SDK v2 we don't need to parse anything from the request, just capture it.